relay-base-schema/src/metrics/name.rs

```diff
@@ -15,6 +15,36 @@ use crate::metrics::MetricNamespace;
 pub struct MetricName(Arc<str>);
 
 impl MetricName {
+    /// Extracts the type from a well formed MRI.
+    ///
+    /// If the contained metric name is not a well formed MRI this function returns `None`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use relay_base_schema::metrics::{MetricName, MetricType};
+    ///
+    /// let name = MetricName::from("cfoo");
+    /// assert!(name.try_type().is_none());
+    /// let name = MetricName::from("c:custom/foo@none");
+    /// assert_eq!(name.try_type(), Some(MetricType::Counter));
+    /// let name = MetricName::from("d:custom/foo@none");
+    /// assert_eq!(name.try_type(), Some(MetricType::Distribution));
+    /// let name = MetricName::from("s:custom/foo@none");
+    /// assert_eq!(name.try_type(), Some(MetricType::Set));
+    /// let name = MetricName::from("g:custom/foo@none");
+    /// assert_eq!(name.try_type(), Some(MetricType::Gauge));
+    /// ```
+    pub fn try_type(&self) -> Option<MetricType> {
+        match self.0.as_bytes().get(..2) {
```

Comment:
Why `as_bytes()`? Is it just to skip the boundary check in [`String::get`](https://doc.rust-lang.org/std/string/struct.String.html#examples-52)?

Reply:
Yes, no need to check for the boundary, we know the first 2 characters must be ascii.